repo/content/committed_read_manager.go

```diff
@@ -336,9 +336,15 @@ func (sm *SharedManager) addRef() {
 
 // release removes a reference to the shared manager and destroys it if no more references are remaining.
 func (sm *SharedManager) release(ctx context.Context) error {
+	if atomic.LoadInt32(&sm.closed) != 0 {
```

Comment:
This is a bug fix, right? Can we merge it ahead by itself in a separate PR?

Reply:
done